### ISSUE TITLE
M2 macOS Ventura: ERR_TUNNEL_CONNECTION_FAILED when configuring system proxies

### ISSUE THREAD

**Issue by Nezteb** [mitmproxy/mitmproxy#5946]
#### Problem Description
Despite following the instructions as closely as possible, I'm unable to get any actual traffic routed through `mitmproxy` or `mitmweb` when using it as the system proxy on my network. The proxy returns `502` or `ERR_TUNNEL_CONNECTION_FAILED` errors every time.

#### Steps to reproduce the behavior:
1. Start `mitmweb` (with no other special options)

2. Set up CA cert using CLI: `sudo security add-trusted-cert -d -p ssl -p basic -k /Library/Keychains/System.keychain ~/.mitmproxy/mitmproxy-ca-cert.pem`
<img width="649" alt="Screenshot 2023-02-23 at 8 45 07 AM" src="https://user-images.githubusercontent.com/3588798/220975249-3b816f98-f79f-4177-a3e7-e964338f6e37.png">

3. Configure macOS proxy:
<img width="445" alt="Screenshot 2023-02-23 at 8 44 11 AM" src="https://user-images.githubusercontent.com/3588798/220975380-05d201c5-bab3-4913-bfc7-f486465707f7.png">

4. Visit http://mitm.it/ and see:
<img width="921" alt="Screenshot 2023-02-23 at 8 55 56 AM" src="https://user-images.githubusercontent.com/3588798/220976763-40c11b75-98bc-49f8-a4f3-9e09c0f78591.png">

5. See web UI output:
<img width="957" alt="Screenshot 2023-02-23 at 8 56 11 AM" src="https://user-images.githubusercontent.com/3588798/220976831-03657773-ded2-4de5-9981-c6d155563f6f.png">

6. Visit https://example.com/ and see: (this is using the Brave browser, which does not have its own proxy config and uses the system proxies as normal)
<img width="954" alt="Screenshot 2023-02-23 at 8 59 59 AM" src="https://user-images.githubusercontent.com/3588798/220977306-0b51811c-1b41-4afb-98c8-09580e2a55eb.png">

7. See error output in CLI:
<img width="1093" alt="Screenshot 2023-02-23 at 8 52 10 AM" src="https://user-images.githubusercontent.com/3588798/220975779-58dc3bac-d267-46cb-867a-1a0d4cde357a.png">

8. Try `curl`:
```
❯ curl --proxy 127.0.0.1:8080 --cacert ~/.mitmproxy/mitmproxy-ca-cert.pem https://example.com/

curl: (56) Received HTTP code 502 from proxy after CONNECT
```

9. Turn off the system proxies from step 3, then try curl again:
```
❯ curl --proxy 127.0.0.1:8080 --cacert ~/.mitmproxy/mitmproxy-ca-cert.pem https://example.com/

<!doctype html>
<html>
<head>
    <title>Example Domain</title>

    <meta charset="utf-8" />
    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type="text/css">
    body {
        background-color: #f0f0f2;
        margin: 0;
        padding: 0;
        font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;

    }
    div {
        width: 600px;
        margin: 5em auto;
        padding: 2em;
        background-color: #fdfdff;
        border-radius: 0.5em;
        box-shadow: 2px 3px 7px 2px rgba(0,0,0,0.02);
    }
    a:link, a:visited {
        color: #38488f;
        text-decoration: none;
    }
    @media (max-width: 700px) {
        div {
            margin: 0 auto;
            width: auto;
        }
    }
    </style>
</head>

<body>
<div>
    <h1>Example Domain</h1>
    <p>This domain is for use in illustrative examples in documents. You may use this
    domain in literature without prior coordination or asking for permission.</p>
    <p><a href="https://www.iana.org/domains/example">More information...</a></p>
</div>
</body>
</html>
```

#### System Information
```
❯ mitmproxy --version
Mitmproxy: 9.0.1
Python:    3.11.2
OpenSSL:   OpenSSL 1.1.1t  7 Feb 2023
Platform:  macOS-13.2.1-arm64-arm-64bit
```

**Comment by mhils** [comment]
Thanks for the detailed write-up! This looks like macOS is recursively redirecting mitmproxy towards mitmproxy again... I assume plaintext connections don't work either? Did this work with previous macOS versions, or is this the first one you are trying? Also, can anyone else on macOS reproduce this?

**Comment by Nezteb** [comment]
> I assume plaintext connections don't work either?

They don't work either, but they _do_ show up in the `mitmweb` interface, whereas HTTPS connections don't show up at all.

<img width="1037" alt="Screenshot 2023-02-23 at 2 07 40 PM" src="https://user-images.githubusercontent.com/3588798/221042304-53c90e5a-329e-4ba2-a1e2-83b3471ff1c9.png">

> Did this work with previous macOS versions, or is this the first one you are trying?

I've used `mitmproxy` for the past four years on two different Intel Macbooks (including macOS Catalina, Big Sur, and Monterey) and it's always worked beautifully, though I no longer have those Macbooks to go back and test with. 😅

**Comment by mhils** [comment]
> They don't work either, but they _do_ show up in the `mitmweb` interface, whereas HTTPS connections don't show up at all.

That makes sense - for plaintext connections we can wait a bit longer until establishing an upstream connection. This supports my previous theory. Do we have anyone else who can reproduce this?

**Comment by mhils** [comment]
Thanks, @KORraNpl! 🍰 @Nezteb, do you by any chance have any other "auto proxy"/redirect/transparent proxy/corp VPN/... software installed that meddles with your traffic?

**Comment by Nezteb** [comment]
I've since tested with both [Charles Proxy](https://www.charlesproxy.com/) and [Proxyman](https://proxyman.io/), both of which seem to work fine with HTTPS. They have macOS helpers that will automatically configure the CA cert trust settings and the network proxy settings, but other than that the process seems the exact same as `mitmproxy`, so I'm not sure why they work but this doesn't. 🤔 

> do you by any chance have any other "auto proxy"/redirect/transparent proxy/corp VPN/... software installed that meddles with your traffic?

Well shoot, apparently I do:
<img width="714" alt="Screenshot 2023-03-01 at 11 03 45 AM" src="https://user-images.githubusercontent.com/3588798/222239602-50ce3534-731e-4bf0-817e-e06e2af27ed5.png">

Previously I'd been making changes directly from the "Wi-Fi" tab of "System Preferences", not this "Network" tab.

Disabling it causes it to re-enable immediately. Luckily I'm able to remove it; we'll see if my IT department cares enough to ping me about it. 😅 

Closing this issue!
